ui-react/packages/atlasmap-provider/src/components/field/field-util.ts

```diff
@@ -0,0 +1,44 @@
+import { ConfigModel } from '../../models/config.model';
+import { getDocDef } from '../document/document-util';
+import { Field } from '../../models/field.model';
+
+/**
+ * Add the specified field to the current mapping.
+ *
+ * @param fieldId - field identifier
+ * @param mappingId - needed for multiple mapping files
+ */
+export function addToMapping(fieldId: string, _mappingId: string): void {
+  const cfg = ConfigModel.getConfig();
+  const [uri, docType, docName, panel, uuid] = fieldId.split(':');
+  const isSource = (panel === 'source');
```

Comment:
I don’t like inferring info like this, it’s brittle. Let’s keep this in mind when we’ll refactor how we deal with the tree

Reply:
agreed.  Also - the AtlasCanvasViewMappings doesn't establish a new mapping when we drop src to target with no existing mappings.  Not sure how it is supposed to create the new mapping at that point so it can call addToMapping.